opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1739,13 +1741,14 @@ namespace Opm
             }
             case Well2::ProducerCMode::THP:
             {
-                std::vector<double> rates(3, 0.0);
-                for (int p = 0; p<np; ++p) {
-                    rates[p] = -well_state.wellRates()[well_index*np + p];
+                well_state.thp()[well_index] = controls.thp_limit;
+                auto bhp = robustSolveBhpAtThpLimitProd(ebos_simulator, summaryState, deferred_logger);
+                if (bhp) {
+                    well_state.bhp()[well_index] = *bhp;
+                } else {
+                    deferred_logger.warning("FAILURE_GETTING_CONVERGED_BHP",
+                                            "Failed to find BHP when switching to THP control for well " + name());
                 }
-                double bhp = calculateBhpFromThp(rates, well, summaryState, deferred_logger);
-                well_state.bhp()[well_index] = bhp;
-                break;
             }
```

Comment:
What happens if(bhp) is false. Do we need to set the bhp to something reasonable. I guess the break still needs to be there?

Reply:
I think letting the bhp stay unchanged is reasonable, more iterations will be performed for this well in any case.

The break should be there, that is a bug! Only saved accidentally because the next case only breaks...